xarray/core/computation.py

```diff
@@ -282,6 +282,7 @@ def apply_dataarray_vfunc(
     *args,
     signature: _UFuncSignature,
     join: JoinOptions = "inner",
+    broadcast: bool = True,
```

Comment:
Sorry for being unclear (again). I don't think we need this in `align` at all.

We should simply be checking `OPTIONS["arithmetic_broadcast"]` in `Variable._binary_op` . The whole business with `align` was a misunderstanding of mine.

Reply:
I will close this PR as it diverged too much from the original wanted behavior